MatrixSDK/MXSession.m

```diff
@@ -1864,27 +1869,34 @@ - (MXHTTPOperation*)joinRoom:(NSString*)roomIdOrAlias
                      success:(void (^)(MXRoom *room))success
                      failure:(void (^)(NSError *error))failure
 {
+    if (!self.identityService)
+    {
+        NSLog(@"[MXSession] Missing identity service");
+        failure(nil);
```

Comment:
We should return an error to display something to the end user, even if this is a technical error.